src/components/checkbox/__tests__/index.spec.tsx

```diff
@@ -24,57 +27,253 @@ describe('Checkbox renderer test', () => {
+  
+      await driver.press();
+  
+      expect(onValueChange).toHaveBeenCalledTimes(1);
+      expect(onValueChange).toHaveBeenCalledWith(true);
+    });
+  
+    it.each`
+    checkboxInitialValue  | checkboxExpectedValue
+    ${false}              | ${true}
+    ${true}               | ${false}
+    `('Send value ($checkboxInitialValue)', async ({checkboxInitialValue, checkboxExpectedValue}: {checkboxInitialValue: boolean; checkboxExpectedValue: boolean}) => {
+      const props = {onValueChange, value: checkboxInitialValue};
+      const component = <TestCase {...props}/>;
+      const driver = new CheckboxDriver({component, testID});
+  
+      await driver.press();
+  
+      expect(onValueChange).toHaveBeenCalledTimes(1);
+      expect(onValueChange).toHaveBeenCalledWith(checkboxExpectedValue);
+    });
+  });
 
-    expect(onValueChange).toHaveBeenCalledTimes(1);
-    expect(onValueChange).toHaveBeenCalledWith(true);
+  describe('Press', () => {
+    it('Multiple clicks', async () => {
+      const props = {onValueChange};
+      const component = <TestCase {...props}/>;
+      const driver = new CheckboxDriver({component, testID});
+  
+      await driver.press();
+      expect(onValueChange).toHaveBeenCalledTimes(1);
+      expect(onValueChange).toHaveBeenCalledWith(true);
+      await driver.press();
+      expect(onValueChange.mock.calls).toEqual([[true], [false]]);
+      await driver.press();
+      expect(onValueChange.mock.calls).toEqual([[true], [false], [true]]);
+    });
+  
+    it.each`
+    checkboxInitialValue
+    ${false}
+    ${true}
+    `('Disabled (value = $checkboxInitialValue)', async ({checkboxInitialValue}: {checkboxInitialValue: boolean}) => {
+      const props = {onValueChange, value: checkboxInitialValue, disabled: true};
+      const component = <TestCase {...props}/>;
+      const driver = new CheckboxDriver({component, testID});
+  
+      await driver.press();
+  
+      expect(onValueChange).not.toHaveBeenCalled();
+    });
   });
 
-  it.each`
-  checkboxInitialValue  | checkboxExpectedValue
-  ${false}              | ${true}
-  ${true}               | ${false}
-  `('Send value ($checkboxInitialValue)', async ({checkboxInitialValue, checkboxExpectedValue}: {checkboxInitialValue: boolean; checkboxExpectedValue: boolean}) => {
-    const props = {onValueChange, value: checkboxInitialValue};
-    const component = <TestCase {...props}/>;
-    const driver = new CheckboxDriver({component, testID});
+  describe('Validation', () => {
+    describe('onChangeValidity', () => {
+      it('should not been called', async () => {
+        const props = {onChangeValidity};
+        const component = <TestCase {...props}/>;
+        const driver = new CheckboxDriver({component, testID});
+        await driver.press();
 
-    await driver.press();
+        expect(onChangeValidity).not.toHaveBeenCalled();
+      });
+      it('should not been called with required prop', async () => {
```

Comment:
This does not make sense IMO, maybe it comes from here:
```
    if (this.state.validationState && prevState.isValid !== this.state.isValid) {

```

Reply:
As long as we don't invoke `checkboxRef.current?.validate()`, even if 'required' is passed, 'onChangeValidity' should not be invoked